WarehouseManagement.Api/Controllers/DeliveryController.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System.Security.Claims;
+using Microsoft.AspNetCore.Mvc;
+using WarehouseManagement.Common.MessageConstants.Keys;
+using WarehouseManagement.Core.Contracts;
+using WarehouseManagement.Core.DTOs;
+using WarehouseManagement.Core.DTOs.Delivery;
+using static WarehouseManagement.Common.MessageConstants.Keys.DeliveryMessageKeys;
+
+namespace WarehouseManagement.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class DeliveryController : ControllerBase
+{
+    private readonly IDeliveryService deliveryService;
+    private readonly IVendorService vendorService;
+    private readonly IMarkerService markerService;
+
+    public DeliveryController(
+        IDeliveryService deliveryService,
+        IVendorService vendorService,
+        IMarkerService markerService
+    )
+    {
+        this.deliveryService = deliveryService;
+        this.vendorService = vendorService;
+        this.markerService = markerService;
+    }
+
+    [HttpGet("{id}")]
+    [ProducesResponseType(200, Type = typeof(DeliveryDto))]
+    [ProducesResponseType(404)]
+    public async Task<IActionResult> GetDelivery(int id)
+    {
+        var model = await deliveryService.GetByIdAsync(id);
+
+        return Ok(model);
+    }
+
+    [HttpGet("all")]
+    [ProducesResponseType(200, Type = typeof(ICollection<DeliveryDto>))]
+    public async Task<IActionResult> GetDeliveries(
+        [FromQuery] PaginationParameters paginationParams
+    )
+    {
+        var model = await deliveryService.GetAllAsync(paginationParams);
+
+        return Ok(model);
+    }
+
+    [HttpPost("add")]
+    [ProducesResponseType(200)]
+    [ProducesResponseType(400)]
+    public async Task<IActionResult> Add(DeliveryFormDto model)
+    {
+        if (!await vendorService.ExistByIdAsync(model.VendorId))
+        {
+            return BadRequest($"{VendorMessageKeys.VendorWithIdNotFound} {model.VendorId}");
+        }
+
+        foreach (var markerId in model.Markers)
+        {
+            if (!await markerService.ExistById(markerId))
```

Comment:
lets fix this up too. put something inside the MarkersService that checks from an array of markers if it exists. I think it would make sense, no?

You can also re-use it inside `edit`

Reply:
done